### PR TITLE
[Impeller] Respect the respect_ctm flag for mask blurred text & absorb entity scale when rendering the 2-pass blur input.

### PR DESCRIPTION
**Pull request by bdero**
Resolves https://github.com/flutter/flutter/issues/142014
Resolves https://github.com/flutter/flutter/issues/150114

This is a combination of:
1. @gaaclarke's [2-pass blur fix](https://github.com/flutter/flutter/issues/150114#issuecomment-2164001044), and
2. a fix for text mask blurs by supporting the "respect_ctm" flag given by `DlMaskBlurFilter` (thanks to @jonahwilliams for recalling this bug).

Let's see those goldens.

### Skia
Screenshot 2024-06-12 at 8 36 41 PM

### Impeller
Screenshot 2024-06-12 at 8 24 41 PM

### Play/DlGoldenTest.GaussianVsRRectBlur/Metal
image

### CODE REVIEW

**Comment by bdero** [comment]
@gaaclarke I'm pushing this PR to see those goldens, but if you want to take this over or add some more testing stuff feel free to just push a new PR and we can merge that. : )

File: impeller/entity/contents/filters/gaussian_blur_filter_contents.cc [hunk 1]
```diff
@@ -379,6 +379,14 @@ std::optional<Rect> GaussianBlurFilterContents::GetFilterCoverage(
   return expanded_source_coverage.TransformBounds(entity.GetTransform());
 }
 
+namespace {
+Vector2 ExtractScale(const Matrix& matrix) {
```
Comment (jonahwilliams) [inline impeller/entity/contents/filters/gaussian_blur_filter_contents.cc:383]: I think we can use Matrix::Decompose for this

**Comment by jonahwilliams** [comment]
Respect

**Comment by gaaclarke** [comment]
I cherry-picked Brandon's `respect_ctm` fix from this pr into this pr: https://github.com/flutter/engine/pull/53377

I'll add some tests and update the coverage tests/math then send it out for review.  Thanks so much brandon and jonah for hunting that flag down.

**Comment by gaaclarke** [comment]
Oh yea, and I think the gaussian blur is long overdue a comments refresh.  I think we are stabilizing enough to build the docs out.

**Comment by bdero** [comment]
Okie dokie, closing this out. 👍

**Pull request closed without merging**